Automattic/jetpack
# Limited Global Styles: Port feature from ETK to `jetpack-mu-wpcom`

Fixes https://github.com/Automattic/dotcom-forge/issues/8170

## Proposed changes:

Ports the Limited Global Styles feature [from ETK](https://github.com/Automattic/wp-calypso/tree/trunk/apps/editing-toolkit/editing-toolkit-plugin/wpcom-global-styles) to `jetpack-mu-wpcom`.

Code has been just copied from ETK with some additional changes:
- Defined a `MU_WPCOM_GLOBAL_STYLES` constant to turn off the feature in the ETK plugin
- Changed tracks logic to use the tools available in Jetpack
- Changed fetching of the Explorer plan name to use the tools available in Jetpack
- Adapted webpack config and enqueueing logic
- Changed text domain of translations to `jetpack-mu-wpcom`
- Adapted `WPCOM_Global_Styles_Test` to use `\WorDBless\BaseTestCase`
- Fixed a few linting issues:
  - Added missing JSDocs
  - Removed a `?int` syntax because it's not compatible with PHP 7.0
  - Changed a capability check to use the `manage_options` cap instead of the `administrator` role

### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply these changes.
- Sandbox your API.
- Test a Free site with GS.
- The Site Editor should show notices when GS are used.
- The Site Editor should show the `Try before you buy` modal.
- The onboarding flow should work without regressions.
- The Launch Banner should work without regressions.
- Event names should be the same as on production.
